aptos-move/block-executor/src/view.rs

```diff
@@ -627,11 +782,154 @@ impl<'a, T: Transaction, X: Executable> SequentialState<'a, T, X> {
     }
 }
 
+impl<'a, T: Transaction, X: Executable> ResourceState<T> for SequentialState<'a, T, X> {
+    fn set_base_value(&self, key: T::Key, value: ValueWithLayout<T::Value>) {
+        self.unsync_map.set_base_value(key, value);
+    }
+
+    fn read_cached_data_by_kind(
+        &self,
+        _txn_idx: TxnIndex,
+        key: &T::Key,
+        target_kind: ReadKind,
+        layout: UnknownOrLayout,
+        patch_base_value: &dyn Fn(&T::Value, Option<&MoveTypeLayout>) -> anyhow::Result<T::Value>,
+    ) -> ReadResult {
+        match self.unsync_map.fetch_data(key) {
+            Some(mut value) => {
+                // If we have a known layout, upgrade RawFromStorage value to Exchanged.
+                if let UnknownOrLayout::Known(layout) = layout {
+                    assert_matches!(
```

Comment:
Btw this panics in single-node-performance benchmark

Reply:
Same here: we can read something and not replace? Why assertion is there?

Reply:
@gelash suggested to add this assertion in one of his comments.

Reply:
Oh😂